src/compiler/types.ts

```diff
@@ -1869,6 +1875,7 @@ namespace ts {
 
     export interface JSDocTypedefTag extends JSDocTag, Declaration {
         kind: SyntaxKind.JSDocTypedefTag;
+        fullName?: JSDocNamespaceDeclaration | Identifier;

```

Comment:
As we discussed, this might be too complex to do now, and can wait until we have a chance to clean up jsdoc parsing all at once.